packages/rulesets/src/spectral/az-arm.ts

```diff
@@ -96,6 +98,102 @@ const ruleset: any = {
         function: falsy,
       },
     },
+    URIContainsSubscriptionId: {
+      description: "Uri for resource group scoped CRUD methods MUST contain a subscriptionId parameter.",
+      message: "{{error}}",
+      severity: "error",
+      resolved: false,
```

Comment:
Why do you need `resolved: false` here? You are using `resolved: true` for other rules that use the same `verifyArmPath` function, so I think this is probably not needed (or wanted).

Reply:
if for verifying swagger path only, we can just lint on the un-resolved swagger, that would be more efficient

Reply:
I don't believe you gain any efficiency in linting the unresolved API definition.  I believe Spectral _always_ computes the resolved doc _once_, before running any rules, and `resolved` simply determines which input is passed to the function.

Do you have performance measurements that suggest otherwise?